ibis/expr/types/core.py

```diff
@@ -46,9 +47,19 @@ class _FixedTextJupyterMixin(JupyterMixin):
         """JupyterMixin adds a spurious newline to text, this fixes the issue."""
 
         def _repr_mimebundle_(self, *args, **kwargs):
-            bundle = super()._repr_mimebundle_(*args, **kwargs)
-            bundle["text/plain"] = bundle["text/plain"].rstrip()
-            return bundle
+            try:
+                bundle = super()._repr_mimebundle_(*args, **kwargs)
+                bundle["text/plain"] = bundle["text/plain"].rstrip()
+                return bundle
+            except Exception as e:
+                # In IPython (but not other REPLs), exceptions inside of
+                # _repr_mimebundle_ are swallowed to allow calling several display
+                # functions and choosing to display the "best" based on some priority.
+                #
+                # This means that exceptions during interactive repr are silently caught.
+                # We can't stop the exception from being swallowed, but at least we can print them.
+                traceback.print_exc()
+                raise e
```

Comment:
```suggestion
            try:
                bundle = super()._repr_mimebundle_(*args, **kwargs)
            except Exception as e:
                # In IPython (but not other REPLs), exceptions inside of
                # _repr_mimebundle_ are swallowed to allow calling several display
                # functions and choosing to display the "best" based on some priority.
                #
                # This means that exceptions during interactive repr are silently caught.
                # We can't stop the exception from being swallowed, but at least we can print them.
                traceback.print_exc()
                raise e
            else:
                bundle["text/plain"] = bundle["text/plain"].rstrip()
                return bundle
```

Do not bundle the code that is catching the exception you care about with other code that can raise an exception that is unrelated to the thing you're trying to handle.

Reply:
If an error is raised within those last two lines (admittedly very very less likely) then Ipython still is gonna swallow the error silently. I wanted to still see that error. If there were some error handler further up the call stack that would catch the more general error, then I'd be all for restricting our try block to as small as possible. But there is no parent handler, so this is the only chance we have

Reply:
@cpcloud what do you think? I'm willing to change if you still think this is the right move.